drivers/iio/frequency/ad9172.c

```
@@ -100,6 +112,71 @@ static unsigned long long ad9172_get_data_clk(struct cf_axi_converter *conv)
 	return div_u64(dac_rate_Hz, st->interpolation);
 }
 
+static int ad9172_link_status_get(struct ad9172_state *st, unsigned long lane_rate_kHz)
+{
+	struct regmap *map = st->map;
+	struct device *dev = regmap_get_device(map);
+	ad917x_jesd_link_stat_t link_status;
+	int ret, i;
+
+	for (i = JESD_LINK_0; i <= JESD_LINK_1; i++) {
+		ret = ad917x_jesd_get_link_status(&st->dac_h, i, &link_status);
+		if (ret != 0) {
+			dev_err(dev,
+				"DAC:MODE:JESD: ERROR : Get Link%d status failed \r\n", i);
+			return -EIO;
+		}
+
+		dev_info(dev, "Link%d code_grp_sync: %x\n", i, link_status.code_grp_sync_stat);
+		dev_info(dev, "Link%d frame_sync_stat: %x\n", i, link_status.frame_sync_stat);
+		dev_info(dev, "Link%d good_checksum_stat: %x\n",
+			i, link_status.good_checksum_stat);
+		dev_info(dev, "Link%d init_lane_sync_stat: %x\n",
+			i, link_status.init_lane_sync_stat);
+		dev_info(dev, "Link%d %d lanes @ %lu kBps\n",
+			i, st->appJesdConfig.jesd_L, lane_rate_kHz);
+
+		if (hweight8(link_status.code_grp_sync_stat) != st->appJesdConfig.jesd_L ||
+			link_status.code_grp_sync_stat != link_status.frame_sync_stat ||
+			link_status.code_grp_sync_stat != link_status.init_lane_sync_stat)
+			return -EFAULT;
+
+		if (!st->jesd_dual_link_mode)
+			break;
+	}
+
+	return 0;
+}
+
+static int ad9172_finalize_setup(struct ad9172_state *st)
+{
+	ad917x_handle_t *ad917x_h = &st->dac_h;
+	int ret;
+	u8 dac_mask, chan_mask;
+
+	if (st->jesd_dual_link_mode || st->interpolation == 1)
+		dac_mask = AD917X_DAC0 | AD917X_DAC1;
+	else
+		dac_mask = AD917X_DAC0;
+
+	if (st->interpolation > 1) {
+		chan_mask = GENMASK(st->appJesdConfig.jesd_M / 2, 0);
+		ret = ad917x_set_page_idx(ad917x_h, AD917X_DAC_NONE, chan_mask);
+
+		ret = ad917x_set_channel_gain(ad917x_h, 2048); /* GAIN = 1 */
```

above calls are returning but ignoring error codes...

---

Again copy paste